posts/Rust-1.86.0.md

```diff
@@ -0,0 +1,163 @@
+
+If you have a previous version of Rust installed via `rustup`, you can get 1.86.0 with:
+
+```console
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.86.0](https://doc.rust-lang.org/stable/releases.html#version-1860-2025-04-03).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.86.0 stable
+
+### Trait upcasting
+
+This release includes a long awaited feature — the ability to upcast trait objects.
+If a trait has a [supertrait](https://doc.rust-lang.org/reference/items/traits.html#supertraits) you can coerce a reference to said trait object to a reference to a trait object of the super trait:
+
+```rust
+trait Trait: Supertrait {}
+trait Supertrait {}
+
+fn upcast(x: &dyn Trait) -> &dyn Supertrait {
+    x
+}
+```
+
+The same would work with any other kind of (smart)-pointer, like `Arc<dyn Trait> -> Arc<dyn Supertrait>` or `*const dyn Trait -> *const dyn Supertrait`.
+
+Previously this would require a workaround in the form of an `upcast` method in the `Trait` itself (and that would work only for one kind of reference/pointer). Such workarounds are not necessary anymore. 
+
+Note that this adds a new _safety invariant_  to raw pointers — if a pointer points to a trait object, vtable must be valid for that trait. This means that "leaking" a raw pointer to a trait object with an invalid vtable into safe code may lead to undefined behavior.
+
+Trait upcasting may be especially useful with the `Any` trait, as it allows upcasting your trait object to `dyn Any` to call the downcast methods. Whereas before you'd have to write workarounds for this or use external crates.
+
+```rust
+use std::any::Any;
+
+trait MyAny: Any {}
+
+impl dyn MyAny {
+    fn downcast_ref<T>(&self) -> Option<&T> {
+        (self as &dyn Any).downcast_ref()
+    }
+}
+```
+
+You can [learn more about trait upcasting in the Rust reference](https://doc.rust-lang.org/reference/type-coercions.html#unsized-coercions).
+
+### `HashMap`s and slices now support indexing multiple elements mutably
+
+The borrow checker will prevent simaltaneous usage of references returned from `get_mut` methods. To get around this the standard library now provides a `get_disjoint_mut` helper on slices and `HashMap` to retrieve mutable references to multiple elements simaltaneously. See the following example taken from the API docs of [`slice::get_disjoint_mut`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.get_disjoint_mut):
```

Comment:
“The borrow checker will prevent…” could be read as saying that in this and future versions, it will *start to* prevent something. Also, “get around this” sounds a bit sketchy. How about:

```suggestion
The borrow checker prevents simultaneous usage of references obtained from more than one call to `get_mut` methods. To allow simultaneous mutable access to multiple `HashMap` or slice entries, the standard library now provides `get_disjoint_mut` methods to obtain multiple references in a single operation. See the following example taken from the API docs of [`slice::get_disjoint_mut`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.get_disjoint_mut):
```